clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1054,17 +1056,23 @@ static void buildArgTys(ASTContext &Context, CXXRecordDecl *KernelObj,
     QualType ArgTy = Fld->getType();
     if (Util::isSyclAccessorType(ArgTy) || Util::isSyclSamplerType(ArgTy)) {
       createSpecialSYCLObjParamDesc(Fld, ArgTy);
-    } else if (ArgTy->isStructureOrClassType()) {
+    } else if (!ArgTy->isStandardLayoutType()) {
       // SYCL v1.2.1 s4.8.10 p5:
       // C++ non-standard layout values must not be passed as arguments to a
       // kernel that is compiled for a device.
-      if (!ArgTy->isStandardLayoutType()) {
-        const DeclaratorDecl *V =
-            Cpt ? cast<DeclaratorDecl>(Cpt->getCapturedVar())
-                : cast<DeclaratorDecl>(Fld);
-        KernelObj->getASTContext().getDiagnostics().Report(
-            V->getLocation(), diag::err_sycl_non_std_layout_type);
-      }
+      const auto& DiagLocation = Cpt ?
+          Cpt->getLocation() :
+          cast<DeclaratorDecl>(Fld)->getLocation();
+
+      KernelObj->getASTContext().getDiagnostics().Report(
```

Comment:
Why is this going through KernelObj?  We already have the Context as a parameter, right?

Reply:
Now sure why it was done so. Replaced with Context parameter as they're the same. Thanks.